src/Storages/ObjectStorage/S3/Configuration.cpp

```diff
@@ -732,7 +683,10 @@ void StorageS3Configuration::addStructureAndFormatToArgsIfNeeded(
             count -= key_value_asts.size();
         }
 
-        if (count == 0 || count > getMaxNumberOfArguments())
+        if (!count)
+            return;
```

Comment:
What about making `path` a required argument in this case? Currently we use `String path = object_storage_disk->getObjectsKeyPrefix();`, but if you have many table engines sharing the same disk, it will be more convenient to split them by different prefixes. Moreover, if we want to use the same disk for storages as well as for "storage configuration policy", then prefix is a must.